src/Adapter/React/PodOutputStream.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Kubernetes\Client\Adapter\React;
+
+use Evenement\EventEmitter;
+use GuzzleHttp\Promise\PromiseInterface;
+use Kubernetes\Client\Adapter\Http\HttpConnector;
+use Kubernetes\Client\Adapter\Http\HttpNamespaceClient;
+use Kubernetes\Client\Model\Pod;
+use Kubernetes\Client\Model\PodAwareStatusProvider;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\Timer\TimerInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\Util;
+use React\Stream\WritableStreamInterface;
+
+/**
+ * Provide a stream of raw output for the given pod.
+ */
+class PodOutputStream extends EventEmitter implements ReadableStreamInterface
+{
+    /**
+     * @var LoopInterface
+     */
+    private $loop;
+
+    /**
+     * @var bool
+     */
+    private $closed = false;
+
+    /**
+     * @var HttpConnector
+     */
+    private $httpConnector;
+
+    /**
+     * @var float
+     */
+    private $pollInterval;
+
+    /**
+     * @var TimerInterface
+     */
+    private $timer;
+
+    /**
+     * @var PromiseInterface
+     */
+    private $httpPromise;
+
+    /**
+     * @var Pod
+     */
+    private $pod;
+
+    /**
+     * @var HttpNamespaceClient
+     */
+    private $namespaceClient;
+
+    /**
+     * @var PodAwareStatusProvider
+     */
+    private $statusProvider;
+
+    public function __construct(
+        Pod $pod,
+        LoopInterface $loop,
+        HttpConnector $httpConnector,
+        HttpNamespaceClient $namespaceClient,
+        PodAwareStatusProvider $statusProvider,
+        $pollInterval = 0.1
+    ) {
+        $this->loop = $loop;
+        $this->httpConnector = $httpConnector;
+        $this->pollInterval = $pollInterval;
+        $this->pod = $pod;
+        $this->namespaceClient = $namespaceClient;
+        $this->statusProvider = $statusProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isReadable()
+    {
+        return !$this->closed;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function pause()
+    {
+        $this->timer->cancel();
+        if (PromiseInterface::PENDING === $this->httpPromise->getState()) {
```

Comment:
Could you comment out what's the point of this, I don't get it? Maybe write a spec for this?

Reply:
The `pause()` method can be called any time by React when the stream buffer is full and the other side cannot read more data. Because the data comes from an asynchronous Guzzle HTTP call, the promise has to be rejected to stop the streaming. When `pause()` is called right after the promise is fulfilled, calling reject on a fulfilled promise will throw an exception.

I will write spec for this.

Reply:
Hmm. Looks like it's not needed any more, because the the wait function is called in the `resume()` method.